Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -329,148 +249,64 @@ std::string LeptonTriggerSFProducer::GetProducerId() const { return "LeptonTrigg
+        LOG(DEBUG) << "Apply Variation: " << settings.GetLeptonSFVariation();
+        double sf1 = 1 / this->GetScaleFactor(0, *lep1);
+        double sf2 = 1 / this->GetScaleFactor(0, *lep2);
+        product.m_weights["zl1" + weightName] = sf1;
+        product.m_weights["zl2" + weightName] = sf2;
+        product.m_weights["lepton" + weightName] =
+            1 / (1 - (1 - GetScaleFactor(0, *lep1)) * (1 - GetScaleFactor(0, *lep2)));
+        LOG(DEBUG) << "Lepton1 SF: " << sf1 << ", Lepton2 SF: " << sf2;
+        LOG(DEBUG) << "Total SF: " << product.m_weights.at("lepton" + weightName);
         if (settings.GetLeptonSFVariation() == true) {
-            product.m_weights["leptonTriggerSFWeightUp"] =
-                1/(1-
-                  (1-GetScaleFactor(1, *lep1) + 0.005*GetScaleFactor(0, *lep1))
-                * (1-GetScaleFactor(1, *lep2) + 0.005*GetScaleFactor(0, *lep2))
-                );
-            product.m_weights["leptonTriggerSFWeightDown"] =
-                1/(1-
-                  (1-GetScaleFactor(-1, *lep1) - 0.005*GetScaleFactor(0, *lep1))
-                * (1-GetScaleFactor(-1, *lep2) - 0.005*GetScaleFactor(0, *lep2))
-                );
-        }
-    }
-    else {
-        product.m_weights["leptonTriggerSFWeight"] = 0;
-        if (settings.GetLeptonSFVariation() == true) {
-            product.m_weights["leptonTriggerSFWeightUp"] = 0;
-            product.m_weights["leptonTriggerSFWeightDown"] = 0;
+            // up has -1 since it's inside the denominator, down +1
+            double sf1u = 1 / this->GetScaleFactor(-1, *lep1);
+            double sf1d = 1 / this->GetScaleFactor(+1, *lep1);
+            double sf2u = 1 / this->GetScaleFactor(-1, *lep2);
+            double sf2d = 1 / this->GetScaleFactor(+1, *lep2);
+            product.m_weights["zl1" + weightName + "Up"] = sf1u;
+            product.m_weights["zl1" + weightName + "Down"] = sf1d;
+            product.m_weights["zl2" + weightName + "Up"] = sf2u;
+            product.m_weights["zl2" + weightName + "Down"] = sf2d;
+            // proper error propagation needed
+            product.m_weights["lepton" + weightName + "Up"] =
+                1 / (1 - (1 - GetScaleFactor(1, *lep1) + 0.005 * GetScaleFactor(0, *lep1)) *
+                             (1 - GetScaleFactor(1, *lep2) + 0.005 * GetScaleFactor(0, *lep2)));
+            product.m_weights["lepton" + weightName + "Down"] =
+                1 / (1 - (1 - GetScaleFactor(-1, *lep1) - 0.005 * GetScaleFactor(0, *lep1)) *
+                             (1 - GetScaleFactor(-1, *lep2) - 0.005 * GetScaleFactor(0, *lep2)));
+            if (settings.GetDebugVerbosity() > 1) {
+                LOG(DEBUG) << "Lepton1 SF up: " << sf1u << ", Lepton1 SF down: " << sf1d;
+                LOG(DEBUG) << "Lepton2 SF up: " << sf2u << ", Lepton2 SF down: " << sf2d;
+                LOG(DEBUG) << "Total SF up: " << product.m_weights.at("lepton" + weightName + "Up")
```

Comment:
You could think to temporarily save the value inside this scope to avoid searching for the value in the map again. This might save some unnecessary instructions for the price of temporary acquiring a few additional bytes in memory.